src/Resources/Resources.Test/ScenarioTests/TagTests.ps1

```diff
@@ -0,0 +1,282 @@
+﻿# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+Utility method to test CreeteOrUpdate for Tags within tracked resources and subscription.
+#>
+function Test-TagCreateOrUpdateWithResourceIdParams($resourceId)
+{
+    # Setup
+    $expected = @{"key1"="value1"; "key2"="value2";}
+
+    try
+    {
+        # Test
+        $res = New-AzTag -ResourceId $resourceId -Tag $expected
+        [hashtable]$actual = $res.Properties.TagsProperty
+
+        # Assert
+        Assert-True { AreHashtableEqual $expected $actual }
+    }
+    finally
+    {
+        # Cleanup
+        Remove-AzTag -ResourceId $resourceId
+    } 
+}
+
+<#
+.SYNOPSIS
+Tests creating or updating tags on subscription.
+#>
+function Test-TagCreateOrUpdateWithResourceIdParamsForSubscription
+{
+    # Setup
+    # $subs = Get-AzSubscription
+    # $subId = $subs[0].Id
+    $subscriptionId = "b9f138a1-1d64-4108-8413-9ea3be1c1b2d"
+    $resourceId = "/subscriptions/" + $subscriptionId
+
+    Test-TagCreateOrUpdateWithResourceIdParams $resourceId  
+}
+
+<#
+.SYNOPSIS
+Tests creating or updating tags on tracked resource.
+#>
+function Test-TagCreateOrUpdateWithResourceIdParamsForResource
+{
+    # Setup
+    $resourceId = "/subscriptions/b9f138a1-1d64-4108-8413-9ea3be1c1b2d/resourcegroups/TagsApiSDK/providers/Microsoft.Compute/virtualMachines/TagTestVM";
```

Comment:
These hard coded subscription ids and resource group names are bad for tests: next time if someone else tries to re-record these tests, they'll need to replace them.

The suggestion is to create resource groups and resources needed in each case.

Reply:
@isra-fel, thanks for the suggestion and I updated the test cases. 
Also, I have a question regarding to subscriptionId:
Since we have get/put/patch/delete tag operations on subscription directly, we need to provide subscriptionId for testing as well. How could we get the default subscription? 

I have two things in mind:
1. Hard-code the subscriptionId
2. could we use something like: 
    $subs = Get-AzSubscription
    $subscriptionId= "/subscriptions/" + $subs[0].Id

Could you please suggest what to do here or is there any guidance? Many thanks.